src/ad-bidders/prebid/intent-iq/intent-iq.ts

```diff
@@ -92,14 +98,69 @@ export class IntentIQ {
 		externalLogger.log('intentiq report', { report: JSON.stringify(data) });
 	}
 
-	private isEnabled(): boolean {
+	setupPpid(data: IntentIqResponseData) {
+		if (!isIntentIqData(data)) {
+			utils.logger(logGroup, 'no data received');
+			return;
+		}
+
+		if (!this.isEnabled()) {
+			return;
+		}
+
+		const ppid = this.getPpid(data);
+		utils.logger(logGroup, 'ppid', ppid);
+
+		if (ppid) {
+			if (context.get('services.intentIq.ppid.enabled')) {
+				this.setPpid(ppid);
+			}
+			if (context.get('services.intentIq.ppid.tracking.enabled')) {
+				this.trackPpid(ppid);
+			}
+		}
+	}
+
+	isEnabled(): boolean {
 		return (
 			context.get('bidders.prebid.intentIQ') &&
 			context.get('options.trackingOptIn') &&
 			!context.get('options.optOutSale') &&
 			!utils.isCoppaSubject()
 		);
 	}
+
+	getPpid(data: IntentIqData): string | undefined {
```

Comment:
```suggestion
	getPpid(data: IntentIqData): string | null | undefined {
```
or we can `return null` in `catch` and reduce it to `string | null`

Reply:
Great suggestion, I returned `null` in `catch` statement and reduced method return type to `string | undefined`